src/libaktualizr/uptane/uptane_test.cc

```diff
@@ -282,6 +282,44 @@ TEST(Uptane, PutManifestError) {
   EXPECT_EQ(num_events_PutManifestError, 1);
 }
 
+/*
+ * Output a log when connectivity is restored.
+ */
+
+int num_events_LogConnectivityRestored = 0;
+void process_events_LogConnectivityRestored(const std::shared_ptr<event::BaseEvent> &event) {
+  std::cout << event->variant << "\n";
+  if (event->variant == "ConnectivityRestored") {
+    num_events_LogConnectivityRestored++;
+  }
+}
+
+TEST(Uptane, LogConnectivityRestored) {
+  TemporaryDirectory temp_dir;
+  Config conf("tests/config/basic.toml");
+  conf.storage.path = temp_dir.Path();
+  auto storage = INvStorage::newStorage(conf.storage);
+
+  auto http = std::make_shared<HttpFake>(temp_dir.Path());
+  auto sota_client = std_::make_unique<UptaneTestCommon::TestUptaneClient>(conf, storage, http);
+  EXPECT_NO_THROW(sota_client->initialize());
+  EXPECT_TRUE(sota_client->putManifestSimple());
+
+  http = std::make_shared<HttpPutManifestFail>(temp_dir.Path());
+  sota_client = std_::make_unique<UptaneTestCommon::TestUptaneClient>(conf, storage, http);
+  EXPECT_NO_THROW(sota_client->initialize());
+  EXPECT_FALSE(sota_client->putManifestSimple());
+
+  http = std::make_shared<HttpFake>(temp_dir.Path());
+  auto events_channel = std::make_shared<event::Channel>();
+  std::function<void(std::shared_ptr<event::BaseEvent> event)> f_cb = process_events_LogConnectivityRestored;
+  events_channel->connect(f_cb);
+  sota_client = std_::make_unique<UptaneTestCommon::TestUptaneClient>(conf, storage, http, events_channel);
+  EXPECT_NO_THROW(sota_client->initialize());
+  EXPECT_TRUE(sota_client->putManifestSimple());
+  EXPECT_EQ(num_events_LogConnectivityRestored, 1);
+}
```

Comment:
This general test format looks fine, although capturing logging output is slightly challenging, and I'd suggest putting this in `uptane_network_test.cc`, although to be honest this isn't really Uptane-specific.